packages/experimental/Badge/src/colorHelper.ts

```diff
@@ -0,0 +1,209 @@
+      color: colorDark,
+      iconColor: colorDark,
+    },
+    hc: {
+      hcBackground,
+      hcColor,
+      hcBorderColor,
+    },
+  });
+}
+
+/**
+ * A function which returns object of props depending on colors and theme.
+ * @param colors object
+ * @param theme
+ * @returns object of props - color, iconColor and borderColor
+ */
+export function getOutlineColorProps(
+  colors: BadgeColors,
+  theme: Theme,
+  getProps?: (theme: Theme, themeProps: ThemeProps) => Record<string, unknown>,
+) {
+  const { color, colorDark } = colors;
+  const borderColorDark = colors.borderColorDark || colorDark;
+  const getThemeProps = getProps || getDefaultProps;
+  return getThemeProps(theme, {
+    light: {
+      color: color,
+      iconColor: color,
+      borderColor: color,
+    },
+    dark: {
+      color: colorDark,
+      iconColor: colorDark,
+      borderColor: borderColorDark,
+    },
+    hc: getHCProps(theme),
+  });
+}
+
+/**
+ * A function which returns object of props depending on colors and theme.
+ * @param colors object
+ * @param theme
+ * @returns object of props - backgroundColor, color, iconColor and borderColor
+ */
+export function getTintColorProps(
+  colors: BadgeColors,
+  theme: Theme,
+  getProps?: (theme: Theme, themeProps: ThemeProps) => Record<string, unknown>,
+) {
+  const { backgroundColor, color, borderColor, backgroundColorDark, colorDark, borderColorDark } = colors;
+  const getThemeProps = getProps || getDefaultProps;
+  return getThemeProps(theme, {
+    light: {
+      backgroundColor: backgroundColor,
+      color: color,
+      iconColor: color,
+      borderColor: borderColor,
+    },
+    dark: {
+      backgroundColor: backgroundColorDark,
+      color: colorDark,
+      iconColor: colorDark,
+      borderColor: borderColorDark,
+    },
+    hc: getHCProps(theme),
+  });
+}
+
+/**
+ * A function which returns object of props depending on colors and theme.
+ * @param colors object
+ * @param theme
+ * @returns object of props - color and iconColor
+ */
+export function getGhostColorProps(
+  colors: BadgeColors,
+  theme: Theme,
+  getProps?: (theme: Theme, themeProps: ThemeProps) => Record<string, unknown>,
+) {
+  const { color, colorDark } = colors;
+  const getThemeProps = getProps || getDefaultProps;
+  return getThemeProps(theme, {
+    light: {
+      color: color,
+      iconColor: color,
+    },
+    dark: {
+      color: colorDark,
+      iconColor: colorDark,
+    },
+    hc: getHCProps(theme),
+  });
+}
+
+export function getWin32Props(theme: Theme, themeProps: ThemeProps) {
+  const themeAppearance = theme.name;
+  switch (themeAppearance) {
+    case 'White':
+    case 'Colorful':
+    default:
+      return {
+        ...themeProps.light,
+      };
+    case 'DarkGray':
+    case 'Black':
+      return {
+        ...themeProps.dark,
+      };
+    case 'HighContrast':
+      return getHCProps(theme, {
+        ...themeProps.hc,
+      });
+  }
+}
```

Comment:
Nit this could go into a .win32.ts file so that win32 logic stays out of other platforms.